### ISSUE TITLE
Markdown cell <img> doesn't display Base64 Data URI

### ISSUE THREAD

**Issue by DavidJVitale** [jupyterlab/jupyterlab#5183]
In the Classic Jupyter Notebook experience, creating a markdown cell with the following base64 encoded data URI will display the image like so:

```
 <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAEAAAABACAMAAACdt4HsAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAMAUExURQAAAPrCj9lJH3V1d/Tz8+N4WFJRU8LCw/SEGKCgoignKeLi4uqbg/WKIudpK/XOwj49P4yMjmFhY/efWfCIT/7+/tPT1La1tu15OuJpO/fZ0PK1m/SKO/PEtvObaONjK+iPdEhHSeFrR+l6SPS0kzU0Nm5tb/3w5Y+Qk/jf2H5+geyKXPeqa9xVLO12NKSlpujo6faTO9vb2/OBPMvLzOZvO/SJM/vOqlZVV99gNfnEpLi4uvzexO1yLNxXMGloa/ebS/jOtvK9rf307vOUWTIyNPN8LPvh1P3s3vOBMu2jjPSYYO2dfpeYm01MTvm6hPrcy/WRNOaCYfB+OvWUQvejUt9iOeVqM/W8n66vrzk5O/WLK95aLNbX2PSDK0FAQvepdIaGiKqqqvaYPOzs7etyNPelWfvXu729vfzhzObm5vzx7O2QZo6OkOx8Q4GBg+p0Oe6qkVpaXHl6e/vLny8tMGVlZ+V8Vvvt6EVERm9wc/aWUvzn1OJwTfz6+PfLtfrm4OFcKfOjd+mIYl1dXt1aNElJS/Kphvm7ivro4fvUs+BiMvC2o/J7M+mVfHFxcvbSxpOTlPmzdPrNpNpPJ/i+muBmO/egZuaFafiwa/rk2/XMvPB2LelrLN1dOPiiXvrEk5mZmuFuS/iub+JzUvvRq/jd1OmCVfioXeiGYSspLPaZU/nYx/PHuVBPUJCPkDAvMTc3OfzZufS8o3BvcXd3eEA/QVhXWQAAAMwAM8wAZswAmcwAzMwA/8wzAMwzM8wzZswzmcwzzMwz/8xmAMxmM8xmZsxmmcxmzMxm/8yZAMyZM8yZZsyZmcyZzMyZ/8zMAMzMM8zMZszMmczMzMzM/8z/AMz/M8z/Zsz/mcz/zMz///8AAP8AM/8AZv8Amf8AzP8A//8zAP8zM/8zZv8zmf8zzP8z//9mAP9mM/9mZv9mmf9mzP9m//+ZAP+ZM/+ZZv+Zmf+ZzP+Z///MAP/MM//MZv/Mmf/MzP/M////AP//M///Zv//mf//zP///1nuoWgAAAC5dFJOUwD///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////8AZ7UcfAAAAAlwSFlzAAAOwwAADsMBx2+oZAAABLBJREFUWEftlmtbG0UUgLVsEzHFAvVsmlKNlwJeGiAhCaWAoXJrCqVELjEXgsS0hltDE4miYW0tRFpMkWLRYkForddKS63695zZPUl2NkkD+jz6xfdLzszOvDszmZ05T8A/5H/Bvy0whyJmDFPsRjAYDcfjJiwk2Y3AH42G3dVYSLIbQbA0esytxkKS3IJY3fw7X3V2Pv/pkU9OSjXcUri7RArT5BDE5n+x+trbE4nEraa1tcnXy66I1SblCuQQNB61+qyHKbW1vQcPTk469ZauVXyoIIug8rrVI/YmoMBiseh1FwRswJApeNjnMRqxv0xg0V364HtsIidDsOFxGFFgtdsTiaqqKhTovAO/YiMZKQEXmR0lP/Pk9aS/1UcWoe/j1s7W1mfPT67p9U4i0NV735Qay0gJAqopHuBJ2t/qMX57pm7kZIv4QBhbvfdb1w/OSwsrK/WG58Q6Gekp8GSb3/UYHR7jnbv4x8vpKPtpwTuwPncIy0mYNYg5HJ7mjSy9JQ51DRjeuM1hCWEEPxYZN6Rx52D12tyefRgjcsHyUzdzvj3JFzN72EnIBC3vbmD0OLiv92IkIRO8v4xBHvbtx0AkLRBiGOSF2ZDMIv4ddigIaDDIYIeCxd8xyCAtMAWA48kuCZAdrQlA0FVD6uiLOV4whW1m8qVwLj+phEHg0wdbSsAV+8HcRk4cVzEHoWjBrGpbDfdJDPfjQuiP+JYf+O6obcgMnGrYHcFeSkE5FQxxsPS0CyBSzI0+aACILwHYyBQE1SbApgqErRf59NmSFgwRwTYKhlWkZrRNDTYb8G4yDxt5pdmtVaurpzQwTUQpsgtCUVpDxqQu50poTAUN7sWKioqCGmGaDC8FI+DFNaAjCJMaDRmBEC8Ma0kcpSPYlj5Ebtov/kowgkB5CWiOxTmY+FkLwiIJYNP9J+1WUUpadC9yIHBiyzSMAAqntsLDW2QEW7ZZ1TT5PyEwNUGfBsvjQRjsjpd2L+UUuOlQee0okGmEokIwKA5YXEKCSUt2gGDWmkmtRpqKREpQ2CbbrSG6BhRTccZdpiApeNQmH1doVvrlHzzKepvISArMzNdSg6Uaso/ykJoCwNV8L0OufIOBiFxwc0eGi3tfxUhEJoD+/rxnKsD+uWsYScgFI0XGfMfi2CnD+hjGEnIBnCnynHuIcTa4z28bvlRcj4xA6CdX07nlSiwqWL0wY1g3vIClJIwAYkYHuRz77izHFOvJ9ZRdXvHWrxi+U95crAAaHTQ78PiszUfnl9/r6Yh1jNw78fZn4y879Qs63Yr3pYvYMIVCAI1GazJB8Pns9t5EFUmynnE6aYKh885kJkpKAYw00xRBzFAItb2pFMei8771ITaSkSGAlusea7YcSb9wKttGyxSQLdmMSZ5MoNdf7sHHLNkEINQdIIkilVDB5Nk1vXOc+QBkKAQcT+8Wwsj8gcP29vb2W01NZ8+PH+kQK7PBCkyFE6FhepJRKnuuHr9x4/iJHnbvKmAEAVfh6ZJIJIDFHcEIeH/16ZKQTZnRPxZW4BIFQSzuCEYwSkfw2kfM6ZYPdhHNm6+ECvKfg3JYAZgaXORa2A0Kwe75rwUAfwFcMhPNaZxeKQAAAABJRU5ErkJggg==">
```

<img width="762" alt="screen shot 2018-08-21 at 3 53 21 pm" src="https://user-images.githubusercontent.com/18691334/44425627-654d3b00-a55a-11e8-9141-2c876e9e79ab.png">

However, I have not been able to the same markdown cell with the Data URIs to function in notebooks in JupyterLab. Any thoughts? Is this supported functionality?

**Comment by biermeester** [comment]
It's not rendering on my test branch (which contains #5012) since the `src` attribute is getting stripped. I'll look into it.

**Comment by biermeester** [comment]
Seems that `sanitize-html` runs `src` attribute values through a function called `naughtyHref`. And since base64 encoded data is not a `href`, it's judged to be naughty indeed.

I could remedy this by removing the `src` attribute from the default option `allowedSchemesAppliedToAttributes: [ 'href', 'src', 'cite' ]`.

This option was introduced in version 1.18.0 (See https://www.npmjs.com/package/sanitize-html)

**Comment by blink1073** [comment]
How does caja handle base64 encoded data uris?

**Comment by vidartf** [comment]
As an aside: A possible workaround is to use attachments: https://nbformat.readthedocs.io/en/latest/format_description.html#cell-attachments

**Comment by biermeester** [comment]
Looks like Caja doesn't allow it:

` var ALLOWED_URI_SCHEMES = /^(?:https?|geo|mailto|sms|tel)$/i;`

**Comment by blink1073** [comment]
Ah, here is the relevant comment from the classic notebook code: https://github.com/jupyter/notebook/blob/b8b66332e2023e83d2ee04f83d8814f567e01a4e/notebook/static/base/js/security.js#L34

**Comment by biermeester** [comment]
So, want me to remove the `src` attribute from the from the `allowedSchemesAppliedToAttributes` option?

**Comment by blink1073** [comment]
Would that also allow `src` for other element types?

**Comment by biermeester** [comment]
Yes, in my branch the `src` attribute is allowed for `video`, `audio`, `img` and `input` tags, so none of those would be validated.

They all relate to some form of media, so in my opinion, if embedded data is not a big risk for `img` tags, it probably won't be for the others. (_disclaimer: I'm guessing here!_)

**Comment by blink1073** [comment]
I verified that `src` attributes are allowed for `video`, `img`, and `input` tags in the classic notebook.  I couldn't get `audio` tags to render, but I don't see anything on this [page](https://html5sec.org/) of concern about `audio` tags.

**Comment by ian-r-rose** [comment]
This is fixed by #5012. Thanks @biermeester, that was some high-impact work!
